include/lsst/meas/astrom/ScaledPolynomialTransformFitter.h

```diff
@@ -0,0 +1,275 @@
+     *  assuming the per-data-point uncertainties are the quadrature sum
+     *  of the intrinsic scatter and that source's centroid measurement
+     *  uncertainty.
+     */
+    double updateIntrinsicScatter();
+
+    /**
+     *  Return the current estimate of the intrinsic scatter.
+     */
+    double getIntrinsicScatter() const { return _intrinsicScatter; }
+
+    /**
+     *  Mark outliers in the data catalog using sigma clipping.
+     *
+     *  A data point is declare an outlier if:
+     *   - the offset between the model prediction and the data position,
+     *     weighted by the uncertainty (including inferred intrinsic scatter)
+     *     of that point exceeds ctrl.nSigma
+     *   - the number of points already with larger weighted offsets is less
+     *     than ctrl.nClipMax.
+     *  In addition, the worst (in weighted offset) ctrl.nClipMin data points
+     *  are always rejected.
+     *
+     *  @return A pair of the smallest rejected weighted offset (units of sigma)
+     *          and the number of point rejected.
+     */
+    std::pair<double,size_t> rejectOutliers(OutlierRejectionControl const & ctrl);
+
+    /**
+     *  Return a catalog of data points and model values for diagnostic purposes.
+     *
+     *  The values in the returned catalog should not be modified by the user.
+     *
+     *  For information about the schema, either introspect it programmatically
+     *  or see fromMatches and fromGrid.
+     */
+    afw::table::BaseCatalog const & getData() const { return _data; }
+
+    /**
+     *  Return the best-fit transform
+     *
+     *  If f is a ScaledPolynomialTransformFitter instance, then for an
+     *  arbitrary point p:
+     *  @code
+     *  f.getTransform()(p) == f.getOutputScaling().invert()(f.getPoly()(f.getInputScaling()(p)))
+     *  @endcode
+     */
+    ScaledPolynomialTransform const & getTransform() const { return _transform; }
+
+    /**
+     *  Return the polynomial part of the best-fit transform.
+     */
+    PolynomialTransform const & getPoly() const { return _transform.getPoly(); }
+
+    /**
+     *  Return the input scaling transform that maps input data points to [-1, 1].
+     */
+    afw::geom::AffineTransform const & getInputScaling() const { return _transform.getInputScaling(); }
+
+    /**
+     *  Return the output scaling transform that maps output data points to [-1, 1].
+     */
+    afw::geom::AffineTransform const & getOutputScaling() const { return _outputScaling; }
+
+private:
+
+    class Keys;
+
+    ScaledPolynomialTransformFitter(
+        afw::table::BaseCatalog const & data,
+        Keys const & keys,
+        int maxOrder,
+        double intrinsicScatter,
+        afw::geom::AffineTransform const & inputScaling,
+        afw::geom::AffineTransform const & outputScaling
+    );
+
+    double computeIntrinsicScatter() const;
+
+    Keys const & _keys;  // safe as member because this points to one of two singletons.
```

Comment:
Updated to:
```
    // Normally it's not safe to use a reference as a data member because the
    // class holding it can't control when the referenced object gets
    // destroyed, but this points to one of two singletons (which never get
    // destroyed).
```